mentoring/components/answer.py

```diff
@@ -0,0 +1,268 @@
+        if not self.name:
+            return ''
+
+        student_input = self.get_model_object().student_input
+
+        # Default value can be set from another answer's current value
+        if not student_input and self.default_from:
+            student_input = self.get_model_object(name=self.default_from).student_input
+
+        return student_input
+
+    def fallback_view(self, view_name, context=None):
+        context = context or {}
+        context['self'] = self
+        html = loader.render_template('templates/html/answer_editable.html', context)
+
+        fragment = Fragment(html)
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/answer.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/answer.js'))
+        fragment.initialize_js('AnswerBlock')
+        return fragment
+
+    def submit(self, submission):
+        self.student_input = submission[0]['value'].strip()
+        self.save()
+        log.info(u'Answer submitted for`{}`: "{}"'.format(self.name, self.student_input))
+        return {
+            'student_input': self.student_input,
+            'status': self.status,
+            'weight': self.weight,
+            'score': 1 if self.status == 'correct' else 0,
+        }
+
+    @property
+    def status(self):
+        answer_length_ok = self.student_input
+        if self.min_characters > 0:
+            answer_length_ok = len(self.student_input.strip()) >= self.min_characters
+
+        return 'correct' if answer_length_ok else 'incorrect'
+
+    @property
+    def completed(self):
+        return self.status == 'correct'
+
+    def save(self):
+        """
+        Replicate data changes on the related Django model used for sharing of data accross XBlocks
+        """
+        super(AnswerBlock, self).save()
+
+        student_id = self._get_student_id()
+        if not student_id:
+            return  # save() gets called from the workbench homepage sometimes when there is no student ID
+
+        # Only attempt to locate a model object for this block when the answer has a name
+        if self.name:
+            answer_data = self.get_model_object()
+            if answer_data.student_input != self.student_input:
+                answer_data.student_input = self.student_input
+                answer_data.save()
+
+    @classmethod
+    def get_template(cls, template_id):
+        """
+        Used to interact with Studio's create_xblock method to instantiate pre-defined templates.
+        """
+        # Generate a random 'name' value
+        if template_id == 'studio_default':
+            return {'metadata': {'name': uuid.uuid4().hex[:7]}, 'data': {}}
+        return {'metadata': {}, 'data': {}}
+
+
+class AnswerRecapBlock(AnswerMixin, StudioEditableXBlockMixin, XBlock):
+    """
+    A block that displays an answer previously entered by the student (read-only).
+    """
+    name = String(
```

Comment:
This should be a drop-down of possible choices. I think I mentioned this in the flat comments, but for tracking, placing it here, too.